### PR TITLE
API 7.4

### LINKED ISSUE #4275: [FEATURE] API 7.4
### What kind of feature are you missing? Where do you notice a shortcoming of PTB?

⭐ 

### Important

- Comment here or in the dev group what you want to work on so we can assign you
- Please copy the relevant part of the checklist from
- https://github.com/python-telegram-bot/python-telegram-bot/issues/4275 to your PR and make sure to cover everything :)
- Please base your branch on the [api-7.4](https://github.com/python-telegram-bot/python-telegram-bot/tree/api-7.4) branch
- If this is your first contribution, please be sure to read the [contribution guide](https://docs.python-telegram-bot.org/en/latest/contributing.html) and feel free to reach out with any questions via https://t.me/pythontelegrambotdev

### Describe the solution you'd like

- [x] Added support for payments in [Telegram Stars](https://t.me/BotNews/90) by introducing the new currency “XTR”. (Taken by @harshil21)
- [x] The parameter provider_token of the methods [sendInvoice](https://core.telegram.org/bots/api#sendinvoice) and [createInvoiceLink](https://core.telegram.org/bots/api#createinvoicelink) must be omitted for payments in [Telegram Stars](https://t.me/BotNews/90). (Taken by @harshil21)
- [x] The field provider_token in the class [InputInvoiceMessageContent](https://core.telegram.org/bots/api#inputinvoicemessagecontent) must be omitted for payments in [Telegram Stars](https://t.me/BotNews/90). (Taken by @harshil21)
- [x] Added the method [refundStarPayment](https://core.telegram.org/bots/api#refundstarpayment). (Taken by @harshil21)
- [x] Added the field effect_id to the class [Message](https://core.telegram.org/bots/api#message). (Taken by @harshil21)
- [x] Added the parameter message_effect_id to the methods [sendMessage](https://core.telegram.org/bots/api#sendmessage), [sendPhoto](https://core.telegram.org/bots/api#sendphoto), [sendVideo](https://core.telegram.org/bots/api#sendvideo), [sendAnimation](https://core.telegram.org/bots/api#sendanimation), [sendAudio](https://core.telegram.org/bots/api#sendaudio), [sendDocument](https://core.telegram.org/bots/api#senddocument), [sendSticker](https://core.telegram.org/bots/api#sendsticker), [sendVideoNote](https://core.telegram.org/bots/api#sendvideonote), [sendVoice](https://core.telegram.org/bots/api#sendvoice), [sendLocation](https://core.telegram.org/bots/api#sendlocation), [sendVenue](https://core.telegram.org/bots/api#sendvenue), [sendContact](https://core.telegram.org/bots/api#sendcontact), [sendPoll](https://core.telegram.org/bots/api#sendpoll), [sendDice](https://core.telegram.org/bots/api#senddice), [sendInvoice](https://core.telegram.org/bots/api#sendinvoice), [sendGame](https://core.telegram.org/bots/api#sendgame), and [sendMediaGroup](https://core.telegram.org/bots/api#sendmediagroup). (Taken by @harshil21)
- [x] Added the field show_caption_above_media to the classes [Message](https://core.telegram.org/bots/api#message), [InputMediaAnimation](https://core.telegram.org/bots/api#inputmediaanimation), [InputMediaPhoto](https://core.telegram.org/bots/api#inputmediaphoto), [InputMediaVideo](https://core.telegram.org/bots/api#inputmediavideo), [InlineQueryResultGif](https://core.telegram.org/bots/api#inlinequeryresultgif), [InlineQueryResultMpeg4Gif](https://core.telegram.org/bots/api#inlinequeryresultmpeg4gif), [InlineQueryResultPhoto](https://core.telegram.org/bots/api#inlinequeryresultphoto), [InlineQueryResultVideo](https://core.telegram.org/bots/api#inlinequeryresultvideo), [InlineQueryResultCachedGif](https://core.telegram.org/bots/api#inlinequeryresultcachedgif), [InlineQueryResultCachedMpeg4Gif](https://core.telegram.org/bots/api#inlinequeryresultcachedmpeg4gif), [InlineQueryResultCachedPhoto](https://core.telegram.org/bots/api#inlinequeryresultcachedphoto), and [InlineQueryResultCachedVideo](https://core.telegram.org/bots/api#inlinequeryresultcachedvideo). (Taken by @harshil21)
- [x] Added the parameter show_caption_above_media to the methods [sendAnimation](https://core.telegram.org/bots/api#sendanimation), [sendPhoto](https://core.telegram.org/bots/api#sendphoto), [sendVideo](https://core.telegram.org/bots/api#sendvideo), [copyMessage](https://core.telegram.org/bots/api#copymessage), and [editMessageCaption](https://core.telegram.org/bots/api#editmessagecaption). (Taken by @harshil21)
- [x] Added support for “expandable_blockquote” entities in received messages.
- [x] Added support for “expandable_blockquote” entity parsing in “MarkdownV2” and “HTML” parse modes.
- [x] Allowed to explicitly specify “expandable_blockquote” entities in formatted texts.

### Describe alternatives you've considered

If you want to contribute, you can start from the [Contribution Guide](https://docs.python-telegram-bot.org/en/latest/contributing.html).

### Additional context

Please comment below before start working on any part of the update to keep things in track and avoid double work.

### PR DESCRIPTION
**Pull request by harshil21**
Central PR which will gather all PRs from api-7.4-feature

Closes #4275 .

- [ ] PRS to mention: #4278, #4279, #4280, #4286, #4283, #4285

### Checklist

- [ ] Added ``.. versionadded:: NEXT.VERSION``, ``.. versionchanged:: NEXT.VERSION`` or ``.. deprecated:: NEXT.VERSION`` to the docstrings for user facing changes (for methods/class descriptions, arguments and attributes)
- [ ] Created new or adapted existing unit tests
- [ ] Documented code changes according to the `CSI standard <https://standards.mousepawmedia.com/en/stable/csi.html>`__
- [ ] Added myself alphabetically to ``AUTHORS.rst`` (optional)
- [ ] Added new classes & modules to the docs and all suitable ``__all__`` s
- [ ] Checked the `Stability Policy <https://docs.python-telegram-bot.org/stability_policy.html>`_ in case of deprecations or changes to documented behavior

**If the PR contains API changes (otherwise, you can ignore this passage)**

- [ ] Checked the Bot API specific sections of the [Stability Policy](https://docs.python-telegram-bot.org/stability_policy.html)

- New classes:

   - [ ] Added ``self._id_attrs`` and corresponding documentation
   - [ ] ``__init__`` accepts ``api_kwargs`` as kw-only

- Added new shortcuts:

   - [ ]  In :class:`~telegram.Chat` & :class:`~telegram.User` for all methods that accept ``chat/user_id``
   - [ ]  In :class:`~telegram.Message` for all methods that accept ``chat_id`` and ``message_id``
   - [ ]  For new :class:`~telegram.Message` shortcuts: Added ``do_quote`` argument if methods accepts ``reply_to_message_id``
   - [ ] In :class:`~telegram.CallbackQuery` for all methods that accept either ``chat_id`` and ``message_id`` or ``inline_message_id``

-  If relevant:

   - [ ] Added new constants at :mod:`telegram.constants` and shortcuts to them as class variables
   - [ ] Link new and existing constants in docstrings instead of hard-coded numbers and strings
   - [ ]  Add new message types to :attr:`telegram.Message.effective_attachment`
   - [ ] Added new handlers for new update types
   - [ ] Add the handlers to the warning loop in the :class:`~telegram.ext.ConversationHandler`
   - [ ] Added new filters for new message (sub)types
   - [ ]  Added or updated documentation for the changed class(es) and/or method(s)
   - [ ] Added the new method(s) to ``_extbot.py``
   - [ ] Added or updated ``bot_methods.rst``
   - [x] Updated the Bot API version number in all places: ``README.rst`` and ``README_RAW.rst`` (including the badge), as well as ``telegram.constants.BOT_API_VERSION_INFO``
   - [ ] Added logic for arbitrary callback data in :class:`telegram.ext.ExtBot` for new methods that either accept a ``reply_markup`` in some form or have a return type that is/contains :class:`~telegram.Message`

### CODE REVIEW

**Comment by harshil21** [comment]
not sure why `test_meta` is failing on 3.12 (setuptools is missing somehow), probably some kind of caching issue?

**Comment by Bibo-Joshi** [comment]
@harshil21 Its failing on master as well. Maybe we&#x27;ll have to ensure installing setup tools in the CI

File: telegram/_inline/inlinekeyboardbutton.py [hunk 1]
```diff
@@ -123,11 +124,17 @@ class InlineKeyboardButton(TelegramObject):
             insert the bot's username and the specified inline query in the input field. Not
             supported for messages sent on behalf of a Telegram Business account.
         callback_game (:class:`telegram.CallbackGame`, optional): Description of the game that will
-            be launched when the user presses the button. This type of button **must** always be
-            the **first** button in the first row.
-        pay (:obj:`bool`, optional): Specify :obj:`True`, to send a Pay button. This type of button
-            **must** always be the **first** button in the first row and can only be used in
-            invoice messages.
+            be launched when the user presses the button
```
Comment (Poolitzer) [inline telegram/_inline/inlinekeyboardbutton.py:127]:
```suggestion
            be launched when the user presses the button.
```

File: telegram/_inline/inputinvoicemessagecontent.py [hunk 2]
```diff
@@ -85,20 +92,20 @@ class InputInvoiceMessageContent(InputMessageContent):
...
-            shipping address to complete the order
-        send_phone_number_to_provider (:obj:`bool`, optional): Pass :obj:`True`, if user's phone
-            number should be sent to provider.
-        send_email_to_provider (:obj:`bool`, optional): Pass :obj:`True`, if user's email address
-            should be sent to provider.
-        is_flexible (:obj:`bool`, optional): Pass :obj:`True`, if the final price depends on the
-            shipping method.
+            address to complete the order. Ignored for payments in |tg_stars|.
+        need_shipping_address (:obj:`bool`, optional): Pass :obj:`True`, if you require the
+            user's shipping address to complete the order. Ignored for payments in |tg_stars|
```
Comment (Poolitzer) [inline telegram/_inline/inputinvoicemessagecontent.py:102]:
```suggestion
            user's shipping address to complete the order. Ignored for payments in |tg_stars|.
```

**Review by Poolitzer — CHANGES_REQUESTED** [review]
.

File: telegram/_bot.py [hunk 3]
```diff
@@ -4890,7 +4994,7 @@ async def send_invoice(
                 a maximum tip of US$ 1.45 pass ``max_tip_amount = 145``. See the exp parameter in
                 `currencies.json <https://core.telegram.org/bots/payments/currencies.json>`_, it
                 shows the number of digits past the decimal point for each currency (2 for the
-                majority of currencies). Defaults to ``0``.
+                majority of currencies). Defaults to ``0``. Not supported for payment in |tg_stars|
```
[the remark this answers is not in the document]
Comment (harshil21) [reply telegram/_bot.py:4997]: yea that period takes it over 100 chars that's why I didn't put it lmao

**Pull request merged**